contracts/ManagerProxy.sol

```diff
@@ -4,48 +4,42 @@ import "./ManagerProxyTarget.sol";
 
 
 contract ManagerProxy is ManagerProxyTarget {
-    function ManagerProxy(address _controller, bytes32 _targetContractId) Manager(_controller) {
+    function ManagerProxy(address _controller, bytes32 _targetContractId) public Manager(_controller) {
         targetContractId = _targetContractId;
     }
 
     // Based on https://github.com/AugurProject/augur-core/blob/develop/src/libraries/Delegator.sol
     function() public payable {
-        // Set size of method call result to 32 for now until
-        // we can use the RETURNDATACOPY and RETURNDATASIZE opcodes introduced
-        // in the Byzantium hard fork
-        uint32 size = 32;
         address target = controller.getContract(targetContractId);
         // Target contract must be registered
         require(target > 0);
 
         assembly {
             // Load the free memory pointer at 0x40
             let calldataMemoryOffset := mload(0x40)
-            // Set size of reserved memory space for calldata and method call results
-            // Reserved memory size = max(calldatasize, size) so we can always
-            // use the space for loading calldata and then reusing the space
-            // to load method call results
-            let reservedSize := 0
-            switch gt(calldatasize, size)
-            case 1 {
-                reservedSize := calldatasize
-            } default {
-                reservedSize := size
-            }
-            // Update free memory pointer to after memory space we reserve for calldata and method call results
-            mstore(0x40, add(calldataMemoryOffset, reservedSize))
-
+            // Update free memory pointer to after memory space we reserve for calldata
+            mstore(0x40, add(calldataMemoryOffset, calldatasize))
             // Copy method signature and params of the call to memory
             calldatacopy(calldataMemoryOffset, 0x0, calldatasize)
+
             // Call method on target contract and store result starting at calldataMemoryOffset
-            let ret := delegatecall(gas, target, calldataMemoryOffset, calldatasize, calldataMemoryOffset, size)
+            let ret := delegatecall(gas, target, calldataMemoryOffset, calldatasize, 0, 0)
+
+            // Load the free memory pointer at 0x40
+            let returndataOffset := mload(0x40)
+            // Update free memory pointer to after memory space we reserve for returndata
+            mstore(0x40, add(returndataOffset, returndatasize))
```

Comment:
two nits: use consistent naming for callDataMemoryOffset and returndataOffset, and in one call we use `0x0` and the other just `0`.